compiler/graindoc/docblock.re

```diff
@@ -7,28 +8,41 @@ type param = {
   param_id: string,
   param_type: string,
   param_msg: string,
+  param_loc: Location.t,
```

Comment:
Nit could use `Location.loc('a)` for these? May make some of the pattern matches a bit more consistent rather than having to keep track of the location field of each type :shrug:

Reply:
The `loc` type is good for when you don't need to carry the location information forward. I feel like it's too easy to do a `.txt` and then pass these record types along, and when you realize you don't have the location information you have to go back and thread it through.

That said, I actually don't like having these types here. They're only used in this file and I think named arguments would be just fine. I didn't want to make that opinionated change in this PR though.